assessment/collectors/app_code_collector.go

```diff
@@ -286,14 +309,17 @@ func (m *MigrationCodeSummarizer) InvokeCodeConversion(
 	return llmResponse, nil
 }
 
-func formatQuestionsAndSearchResults(questions []string, searchResults [][]string) string {
+func formatQuestionsAndSearchResults(questions []string, conceptSearchResults [][]string, querySearchResults [][]string) string {
 	formattedString := "Use the following questions and their corresponding answers to guide the code conversions:\n**Clarifying Questions and Potential Solutions:**\n\n"
 
 	for i, question := range questions {
-		if len(searchResults[i]) > 0 {
+		if len(conceptSearchResults[i]) > 0 || len(querySearchResults[i]) > 0 {
```

Comment:
nit: Can we use a clearer name than concept here. I'm assuming this is the non-query code. Maybe just call it the appCode or some equivalent ?